src/background.js

```diff
@@ -29,50 +29,74 @@ function sendVideoUrlToMetubeAndSwitchTab(videoUrl, metubeUrl, tab) {
         if (res.ok === true && res.status === 200) {
             return res.json();
         }
-        // todo fix it
-        alert("error :: code " + res.status);
+        console.log("error :: code" + res.status);
     }).then(function(result) {
-        if (result.status === "ok") {
-            openTab(metubeUrl, tab);
-        } else {
-            // todo fix it
-            alert("error :: " + json);
+        if (result.status !== "ok") {
+            console.log("error :: ", result );
+            if (result.msg.includes('URLError')) {
+                // Go straight to catch block and skip the callback
+                throw new Error('Error when adding URL to MyTube');
```

Comment:
sorry, i can't understand the idea of that construction, can you explain that?

why check result to contain `URLError` and throw Exception?

Reply:
If the currently open tab's URL is rejected by MeTube, we get a response back with the "URLError" string inside it.

Example: If the current tab is say the extensions page, or a website which does not have any video on it at all like github - in these cases, MeTube returns with an error body with "URLError" in it.

In such a case, I throw a new Error so that the callback function is skipped entirely. This is because I want the callback function to be executed only when the "send URL to MeTube" part is successful.

Reply:
i think you overcomplicating this code.
1. code will be more clean if check values types in header of function
2. in chain of promises you need to check the request was ok or not

try to rewrite it into something like that

```
if (typeof callback !== 'function'){
    callback = function(){}
}

fetch(metubeUrl + "/add", {
        ...
    }).then(function(res) {
        if (res.ok === true && res.status === 200) {
            callback()
        }
        console.log("error :: code" + res.status + " :: response :: " + res);
    })
```

Reply:
Thanks for the helpful suggestion!
I have since refactored this code to be similar to what you suggested, but with some changes: 

meTube returns a HTTP 200 OK response code for the case when a known non-video URL is passed to it, so you are forced to inspect the response json body to know if meTube could correctly process the URL that was sent to it or not.
Hence, I've now added handling for that case, while also keeping a catch for any unexpected errors that may happen in this flow too

Here's the demo that it works now:
![sample](https://user-images.githubusercontent.com/4332373/201548117-2b040600-68db-47fe-86e5-f9846d2c2cd6.png)
Here, request 1 is when I clicked on the meTube extension icon from chrome://extensions page and meTube, understandably, did not like the URL to extract any video
For request 2, I clicked on the meTube extension icon from a regular youtube video page, and it worked just fine.